### ISSUE TITLE
Hex Casting

### ISSUE THREAD

**Issue by IcarusWhite** [EnigmaticaModpacks/Enigmatica9#143]
### URL

https://www.curseforge.com/minecraft/mc-mods/hexcasting

### Why would you like the mod added?

"A mod for Forge and Fabric adding stack-based programmable spellcasting, inspired by Psi."

Hex Casting is a magic mod that demands much from players, but rewards them with one of the most powerful and versatile magic systems I've seen. Learning how to cast spells in this mod really feels like learning how to cast magic. Cast on the fly! Make artifacts and share them with friends! [REDACTED]! And even more features are coming with the v1.0 update, coming soon!

For some examples of what a more advanced caster can accomplish, see [this gallery](https://imgur.com/a/oSGl2jX)

Compared to Ars, Hex Casting might seem limited at first. Hex Casting follows the Psi philosophy of design: there are fewer possible effects - you can't freeze someone with a Hex, for example - but with enough preparation, you can make incredible things, like a spell that [builds a castle out of obsidian](https://imgur.com/a/YvQIDvQ), or a [magical quarry spell](https://imgur.com/a/zOcYu9D).

In conclusion, Hex Casting might be compared to a mix of Psi and ComputerCraft: hard to learn, but extremely satisfying and powerful when mastered. I loved Hex Casting in Enigmatica 8, and I hope it can become a part of Enigmatica 9. (Just don't give your true name to someone you don't trust, that never ends well...)

**Comment by MuteTiefling** [comment]
I feel Ars Nouveau covers the magic spell niche sufficiently, to be honest.

**Comment by NielsPilgaard** [comment]
Mkay 👍

**Comment by yoyotam3** [comment]
I'd like to appeal this. Hex Casting might be a spell programming mod like Ars, but it's much more advanced. It's honestly more advanced than Psi, since it has things like conditionals logic, saving spells as variables and running them in other spells etc. It's actually a turing complete language. Plus, unlike Psi, it actually has some things that it is legitimately useful for even in modpacks, like for example the spell Greater Teleport lets you teleport to any coordinates. The mod also has great theming/lore, with a eldritch horror/insanity type of plot where you gradually become corrupted by the knowledge you find and later mechanics involve things like stealing and modifying the brains of villagers to use as components for spell automation. TLDR outside of the basic idea of spell creation it really isn't very similar to Ars Nouveau at all since it's much more advanced, and the programming system is fun to use and the progression and mechanics are cool.

**Comment by NielsPilgaard** [comment]
That does sound very appealing, the problem is that we'd have to fit it into expert mode somehow.

**Comment by yoyotam3** [comment]
I don't feel like every single mod in a pack has to be compulsory for expert mode. For example in e2e gating psi until fairly late and forcing players to get through it in a specific way just meant that the mod was completely useless by the time you got it and most players were only interested in getting past it ASAP. If you absolutely have to fit it into expert as a compulsory mod though, you might be able to gate things behind Mind Flaying recipes?

**Comment by MuteTiefling** [comment]
That tends to be the issue with these types of mods though; they're really powerful out of the gate with little opportunity to gate their more powerful features.

So if you stick them at the start, they make the other mods useless. But if you gate them, they're so late there's no incentive to use them anymore.

I can take a closer look at it in a few days, but my impression so far is that I'm not going to integrate it well into expert, so it would just end up heavily gated or disabled.

**Comment by yoyotam3** [comment]
This mod actually has a lot of opportunity to gate the more powerful features. In order to get scrolls to find the stroke order for Great Spells, which are necessary for the more powerful stuff (greater teleport is necessary for teleporting with coordinates, greater sentinel is necessary to affect anything which is further than your range which i think is 32 blocks?, etc.) you have to find scrolls for them, which you should be able to gate behind other things. Also I should mention that this mod has a cool addon named Hexal that adds various things, though I dont know how much of it is gateable.

**Comment by IcarusWhite** [comment]
I am going to add a couple notes here.
I haven't had time to play as much recently, but:
1. The mod is still under active development. In particular, I believe there is now a "magic elytra" like Ars has, as well as a cheap write-only storage for spells.
2. The spell circles, which are the only way to automate greater spells, require unlocking Mind Flay; however, they could be further gated by changing the recipes for the empty impetus.
3. It is actually possible to automate "normal" spells using something like a Create deployer and an artifact or trinket. In fact, it is now possible for a trinket to refuel itself, allowing for full automation!
4. I believe the API now allows for custom "ingredients" for Mind Flay.
While gating things behind a high skill mod like hex is always risky, I think the mod would add a certain technical magic touch to the pack.
Again, I don't have as much time to play lately, but I thought I'd add my thoughts, since I did start the topic.

**Comment by huntermarchildon** [comment]
I personally would love to have this as an option, even if it ends up very heavily gated in expert mode. I love the idea of this mod and enigmatica 9 together and would love an option to play them together for options on powering myself up or automating things. This mod also is slightly less exploitable than something like psi because there is no direct copy paste feature.

**Comment by huntermarchildon** [comment]
thats a shame to hear, but i understand and i will probably add it to the
pack myself and only do things on tier for expert mode.
